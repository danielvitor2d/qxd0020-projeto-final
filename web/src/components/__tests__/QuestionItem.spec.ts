import { shallowMount } from '@vue/test-utils';
import { describe, expect, it, vi } from 'vitest';

import QuestionItemVue from '../QuestionItem.vue';

describe('QuestionItemVue', () => {
  it('renders correctly', () => {
    const item = {
      id: 1,
      label: 'Test Label',
      course: 'Test Course'
    };

    const onDelete = vi.fn()

    const wrapper = shallowMount(QuestionItemVue, {
      props: {
        item: item,
        onDelete: onDelete
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
