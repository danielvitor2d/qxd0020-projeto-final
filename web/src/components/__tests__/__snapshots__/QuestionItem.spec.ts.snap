// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QuestionItemVue > renders correctly 1`] = `
"<div class=\\"w-full flex flex-row items-center justify-between\\">
  <p class=\\"flex-1\\">Test Label</p>
  <p class=\\"mr-4\\">Test Course</p><button>
    <font-awesome-icon icon=\\"fa-xmark\\"></font-awesome-icon>
  </button>
</div>"
`;
